typhos/display.py

```diff
@@ -26,9 +25,202 @@ class DisplayTypes(enum.IntEnum):
+
+    def generate_context_menu(self):
+        if not self.templates:
+            return
+
+        menu = QtWidgets.QMenu(parent=self)
+        for template in self.templates:
+            def selected(*, template=template):
+                self.template_selected.emit(template)
+
+            action = menu.addAction(template.name)
+            action.triggered.connect(selected)
+        return menu
+
+    def open_context_menu(self, ev):
+        menu = self.generate_context_menu()
+        menu.exec_(self.mapToGlobal(ev.pos()))
+
+
+class TyphosDisplaySwitcher(QtWidgets.QFrame, widgets.TyphosDesignerMixin):
+    """
+    Display switcher button set for use with a Typhos Device Display
+    """
+    template_selected = QtCore.Signal(pathlib.Path)
+
+    icons = {'embedded_screen': 'compress',
+             'detailed_screen': 'braille',
+             'engineering_screen': 'cogs'
+             }
+
+    def __init__(self, parent=None, **kwargs):
+        # Intialize background variable
+        super().__init__(parent=None)
+
+        self.device_display = None
+        self.buttons = {}
+        layout = QtWidgets.QHBoxLayout()
+        self.setLayout(layout)
+        layout.setSpacing(0)
+        layout.setContentsMargins(0, 0, 0, 0)
+
+        self.setContextMenuPolicy(Qt.DefaultContextMenu)
+        self.contextMenuEvent = self.open_context_menu
+
+        if parent:
+            self.setParent(parent)
+
+        self._create_ui()
+
+    def _create_ui(self):
+        layout = self.layout()
+        self.buttons.clear()
+
+        for template_type in DisplayTypes.names:
+            icon = pydm.utilities.IconFont().icon(self.icons[template_type])
+            button = TyphosDisplaySwitcherButton(icon)
+            self.buttons[template_type] = button
+            button.template_selected.connect(self._template_selected)
+            layout.addWidget(button, 0, Qt.AlignRight)
+
+            friendly_name = template_type.replace('_', ' ')
+            button.setToolTip(f'Switch to {friendly_name}')
+
+    def _template_selected(self, template):
+        self.template_selected.emit(template)
+        if self.device_display is not None:
+            self.device_display.force_template = template
+
+    def set_device_display(self, display):
+        self.device_display = display
+
+        for template_type in self.buttons:
+            templates = display.templates.get(template_type, [])
+            self.buttons[template_type].templates = templates
+
+    def add_device(self, device):
+        ...
+
+
+class TyphosDisplayTitle(QtWidgets.QFrame, widgets.TyphosDesignerMixin):
+    """
+    Standardized Typhos Device Display title
+    """
+    def __init__(self, title='${name}', *, show_switcher=True, parent=None):
+        self._show_switcher = show_switcher
+        super().__init__(parent=parent)
+
+        font = QtGui.QFontDatabase.systemFont(QtGui.QFontDatabase.TitleFont)
+        font.setPointSizeF(14.0)
```

Comment:
Can we change the size to be in pixel size? According to https://doc.qt.io/qt-5/highdpi.html it will be an issue due to specification in PointSize.

Reply:
Feel free to commit this change

Reply:
I will get it in